### PR TITLE
improve adapter, add rails submodule for testing

### PR DESCRIPTION
**Pull request by lego**
See CONTRIBUTING.md for a rundown on the changes.

Currently, the adapter has been modified to be compatible with ActiveRecord 5.2.0.beta and CockroachDB v2.0.0-alpha. There are a few minor changes to make the adapter compatible with 5.1 again (e.g. `IndexDefinition.new` takes list args instead of kwargs).

The `rails` repository should be easy to backport test modifications for 5.1.x. Hopefully also 4.2.x, but I don't know how much tests may have changed.

cc: @jordanlewis

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,248 @@
+# Getting started
+
+
+## ActiveRecord adapters and you
+
+There are two repositories for the ActiveRecord adapter. The one you in
```
Comment (jordanlewis) [inline CONTRIBUTING.md]: you're*

File: lib/active_record/connection_adapters/cockroachdb/schema_statements.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
+require 'active_record/connection_adapters/postgresql/schema_statements'
+
+module ActiveRecord
+  module ConnectionAdapters
+    module CockroachDB
+      module SchemaStatements
+        include ActiveRecord::ConnectionAdapters::PostgreSQL::SchemaStatements
+        # NOTE(joey): This was ripped from PostgresSQL::SchemaStatements, with a
+        # slight modification to change setval(string, int, bool) to just
+        # setval(string, int) for CockroachDB compatbility.
```
Comment (bithavoc) [inline lib/active_record/connection_adapters/cockroachdb/schema_statements.rb]: * compatibility

**Comment by jordanlewis** [comment]
Going to merge this and fix up some typos in a separate PR. Praise @lego! 🙇 🙏 🙌

**Comment by bithavoc** [comment]
Awesome!

**Pull request merged**